src/lib.rs

```diff
@@ -197,17 +220,17 @@ impl Watch {
 
         let metadata = metadata();
 
-        self.exclude_path(metadata.target_directory.as_std_path());
+        self.exclude_path(&metadata.target_directory);
```

Comment:
Can you change the `&mut self` of the function argument to `self`?

Right now you are using `&mut self` but this means the things gets mutated during the function and you can still use the struct after the call. This is called ["side effect"](https://en.wikipedia.org/wiki/Side_effect_(computer_science)) and this is wrong because this function shouldn't be called a second time. The method "execute" should execute and not mutate the object (or the object must not be accessible after execution).

Example:

``` rust
pub struct Foo {
    value: Option<u32>, // this field is private
}

impl Foo {
    pub fn new(value: u32) -> Self {
        Self { value: Some(value) }
    }

    pub fn into_value(&mut self) -> u32 {
        self.value.unwrap()
    }
}

fn main() {
    let mut foo = Foo::new(42);
    println!("{}", foo.into_value()); // 42
    println!("{}", foo.into_value()); // panic!
}
```

When you use just `self` in method's argument, the whole object is taken by value. (I personally call that "consumed", I don't know if it's a common technical term.) After the method execution, the object is dropped because it get out of scope.

```rust
fn my_method(self) {
    println!("{:?}", self); // self still exists here
} // self is dropped here with the closing bracket
```

Example fixed:



``` rust
pub struct Foo {
    value: Option<u32>,
}

impl Foo {
    pub fn new(value: u32) -> Self {
        Self { value: Some(value) }
    }

    pub fn into_value(self) -> u32 {
        self.value.unwrap()
    }
}

fn main() {
    let mut foo = Foo::new(42);
    println!("{}", foo.into_value()); // 42
    println!("{}", foo.into_value()); // won't compile!
   // this is because foo has been moved to `into_value()`
}
```